dapps-certification-signature-verification/src/IOHK/Certification/SignatureVerification.hs

```diff
@@ -0,0 +1,132 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+module IOHK.Certification.SignatureVerification
+  ( module CBOR
+  , verifyCIP30Signature
+  , Bech32Address(..)
+  , COSEKey
+  , COSESign1
+  , encodeHex
+  , decodeHex
+  ) where
+
+import IOHK.Certification.SignatureVerification.CBOR as CBOR
+  ( unfoldPayload
+  , unfoldPublicKey
+  , Address(..)
+  , Message(..)
+  , PublicKey(..)
+  , Signature(..)
+  , SignedMessage(..)
+  , UnfoldedPayload(..)
+  )
+import Cardano.Address
+  ( Address, unsafeMkAddress, fromBech32 )
+import Data.Text
+  ( Text )
+import Data.ByteString
+  ( ByteString, empty )
+import Crypto.Error
+  ( CryptoFailable(CryptoPassed, CryptoFailed) )
+import Control.Applicative
+  ((<|>))
+import Control.Monad
+  (unless)
+
+import qualified Data.ByteString.Base16 as Hexa
+import qualified Cardano.Address.Style.Shelley as Shelley
+import qualified Crypto.PubKey.Ed25519 as Ed25519
+import qualified Crypto.Hash.BLAKE2.BLAKE2b as Blake
+
+verifySignature :: PublicKey -> SignedMessage -> Signature -> Either String Bool
```

Comment:
Could we rename this one `verifyEd25519Signature`? What do you think?

Reply:
Done, thanks